src/Adornment/Adornment.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { ReactNode } from 'react'
+
+import { AdornmentWrapper } from './styled'
+import { AdornmentContainer } from './AdornmentContainer'
+import { useAddWidthToPadding, useCloneChildWithPadding } from './hooks'
+
+export enum Position {
+  LEFT = 'Left',
+  RIGHT = 'Right'
+}
+
+export const ADORNMENT_DISPLAY_NAME = 'Adornment'
+export const EXTRA_PADDING = 'extraPadding'
+
+export interface ExtraPadding {
+  [Position.LEFT]: string
+  [Position.RIGHT]: string
+}
+
+type AdornmentProps = {
+  children: ReactNode
+  component: ReactNode
+
+  position?: Position
+  [EXTRA_PADDING]?: ExtraPadding
+}
+
+export const Adornment = ({
+  component,
+  position = Position.LEFT,
+  children,
+  [EXTRA_PADDING]: extraPadding = { [Position.RIGHT]: '0px', [Position.LEFT]: '0px' }
+}: AdornmentProps) => {
+
+  const [extraPaddingWithWidth, setWidth] = useAddWidthToPadding(extraPadding, position)
```

Comment:
how about:
```
const [width, setWidth] = useState(0)
const adornmentRef = useRef()

useEffect(() => setWidth(nodeWidth(adornmentRef.current)), [adornmentRef.current])

const extendedProps = {
  width,
  [position]: inset + 'px'
}

return (
    <AdornmentWrapper>
      {cloneElement(child, { ...child.props, ...extendedProps }}
      <StyledAdornment
        ref={adornmentRef}
        position={position}>
        {component}
      </StyledAdornment>
    </AdornmentWrapper>
  )
```

This way we can get rid of useAddWidthToPadding and useCloneChildWithPadding who both dont adhere to Single responsibility principle.